solana/solana-typegen/src/program/anchor/v0.spec.ts

```
@@ -415,10 +416,27 @@ function fromType(type: IdlType): Type {
             kind: TypeKind.Option,
             type: fromType(type.option),
         }
+    } else if ('coption' in type) {
+        if (type.prefix == 'u8' ) {
+            return {
+                kind: TypeKind.Option,
+                type: fromType(type.coption),
+            }
+        }
+        let d = type.prefix === 'u16' ? 2 : type.prefix === 'u32' ? 4 : 8;
```

should throw on unexpected case